packages/eslint-plugin-pf-codemods/src/rules/helpers/renameSinglePropOnNode.ts

```diff
@@ -0,0 +1,99 @@
+import { JSXAttribute, JSXIdentifier, JSXOpeningElement } from 'estree-jsx';
+import { RenameConfig } from './renamePropsOnNode';
+import { Rule } from 'eslint';
+
+function isRenameConfig(obj: any): obj is RenameConfig {
+  const hasValidObjectType = typeof obj === 'object' && obj !== null;
+
+  const hasValidNewNameType = typeof obj.newName === 'string';
+
+  const isValidMessage =
+    obj.message === undefined ||
+    typeof obj.message === 'string' ||
+    typeof obj.message === 'function';
+
+  const isValidReplace =
+    obj.replace === undefined || typeof obj.replace === 'boolean';
+
+  return (
+    hasValidObjectType &&
+    hasValidNewNameType &&
+    isValidMessage &&
+    isValidReplace
+  );
+}
+
+const getNewName = (
+  removed: boolean,
+  propRename: RenameConfig | string | Record<PropertyKey, never>
+) => {
+  if (removed) return '';
+  return isRenameConfig(propRename) ? propRename.newName : propRename;
+};
+
+const getAction = (
+  removed: boolean,
+  propRename: RenameConfig | string | Record<PropertyKey, never>
+) => {
+  if (removed) return 'removed';
+  return propRename.replace ? 'replaced with ' : 'renamed to ';
+};
+
+const getMessage = (
+  node: JSXOpeningElement,
+  propRename: RenameConfig | string | Record<PropertyKey, never>,
+  defaultMessage: string
+) => {
+  if (!(isRenameConfig(propRename) && propRename.message))
```

Comment:
Nit: IMO it's really easy to misread a notted multi conditional like this and think the logic is actually `!isRenameConfig(propRename) && propRename.message`, I would not both expressions.

```suggestion
  if (!isRenameConfig(propRename) && !propRename.message)
```

Reply:
I might be wrong but these two statements are not equal, are they? 🤔

Reply:
And `Record<PropertyKey, never>` should be a [correct way](https://www.totaltypescript.com/the-empty-object-type-in-typescript) to type an empty object `{}`, which can be passed as a shorthand for removing a prop.

Reply:
Shoot I totally missed these messages, my bad. And yep you're right about the logic!

For the Record type TIL about empty object types, thanks for linking that! I do question the API a bit here, using an empty object for prop removal feels a bit odd for me, but that's something we can discuss later.